### PR TITLE
Code refactoring, fix bugs and change classes to hooks

### PR DESCRIPTION
**Pull request by NikitaBondarevv**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -10,34 +10,28 @@ export const ContextUser = createContext(true);
...
 
-class App extends Component {
-  state = {
-    isAuthenticated: false
-  }
+const App = () => {
+  const [isAuthenticated, setIsAuthenticated] = useState(false)
 
-  async componentDidMount() {
+  useEffect(async () => {
```
Comment (iliatsymbal) [inline index.js:16]: it is forbidden to pass async functions to `useEffect` (see warning in console and make as advised)
image

File: index.js [hunk 2]
```diff
@@ -10,34 +10,28 @@ export const ContextUser = createContext(true);
...
-
-    return (
-      <ContextUser.Provider value={contextValue}>
-        <Header />
-        <Main />
-      </ContextUser.Provider>
-    )
+  const contextValue = {
+    isAuthenticated: isAuthenticated,
+    setUser: setUser
```
Comment (iliatsymbal) [inline index.js:26]: shorthand object properties

File: src/userForm/index.js [hunk 3]
```diff
@@ -1,71 +1,68 @@
...
-
-  fields = [
+export const UserForm = ({ disabledFields }) => {
+  const fields = [
     { label: 'email', reg: /^\w+@\w+\.[a-z]{2,5}$/ },
     { label: 'name', reg: /^[^ ]{3,20}$/ },
     { label: 'surname', reg: /^[^ ]{3,20}$/ },
     { label: 'password', reg: /^[^ ]{6,20}$/, secure: true },
     { label: 'passwordRepeat', reg: /^[^ ]{6,20}$/, secure: true }
   ];
```
Comment (iliatsymbal) [inline src/userForm/index.js:14]: no need to create this array on every render, keep it outside of the component

File: index.js (hunk 1 above)
Comment (NikitaBondarevv) [reply index.js:16]: Thanks, I saw. Fixed it.

**Pull request merged**